File: 13.Linked_List/python/3.InsertNodeHead.py
class Node:
    def __init__(self, data):
        self.data = data
        self.next = None

def insertNodeAtHead(head, data):
    new_node = Node(data)
    new_node.next = head
    return new_node

def printList(head):
    current = head
    while current:
        print(current.data, end=" ")
        current = current.next
    print()

# Start with an empty linked list
head = None

# Simulated input
inputs = [1, 2, 3]

for value in inputs:
    head = insertNodeAtHead(head, value)
    printList(head)

File: 13.Linked_List/javascript/3.InsertNodeHead.js
class Node {
    constructor(data) {
        this.data = data;
        this.next = null;
    }
}

function insertNodeAtHead(head, data) {
    const newNode = new Node(data);
    newNode.next = head;
    return newNode;
}

function printList(head) {
    let current = head;
    let output = '';
    while (current) {
        output += current.data + ' ';
        current = current.next;
    }
    console.log(output.trim());
}

// Start with an empty linked list
let head = null;

// Simulated input
const inputs = [1, 2, 3];

for (let value of inputs) {
    head = insertNodeAtHead(head, value);
    printList(head);
}
